api/router/pubsub/CronBasedEventReceiver.go

```diff
@@ -55,22 +55,23 @@ func NewCronBasedEventReceiverImpl(logger *zap.SugaredLogger, pubsubClient *pubs
 	return cronBasedEventReceiverImpl
 }
 
+//TODO : adhiran : Need to bind to one particular stream. Need to finalise with nishant
 func (impl *CronBasedEventReceiverImpl) Subscribe() error {
-	_, err := impl.pubsubClient.Conn.QueueSubscribe(cronEvents, cronEventsGroup, func(msg *stan.Msg) {
+	_, err := impl.pubsubClient.JetStrCtxt.QueueSubscribe(cronEvents, cronEventsGroup, func(msg *nats.Msg) {
 		impl.logger.Debug("received cron event")
 		defer msg.Ack()
 		event := client.Event{}
 		err := json.Unmarshal([]byte(string(msg.Data)), &event)
 		if err != nil {
-			impl.logger.Errorw("err", "err", err)
+			impl.logger.Errorw("Error while unmarshalling json data", err)
```

Comment:
this may cause error, `impl.logger.Errorw("msg", "key", "value")` is correct way.